docs/develop/typescript/failure-detection.mdx

```diff
@@ -172,79 +160,34 @@ Heartbeat throttling may lead to Cancellation getting delivered later than expec
 Heartbeats can contain a `details` field describing the Activity's current progress.
 If an Activity gets retried, the Activity can access the `details` from the last Heartbeat that was sent to the Temporal Service.
 
-Long-running Activities should Heartbeat their progress back to the Workflow for earlier detection of stalled Activities (with [Heartbeat Timeout](/encyclopedia/detecting-activity-failures#heartbeat-timeout)) and resuming stalled Activities from checkpoints (with Heartbeat details).
+To Heartbeat an Activity Execution in TypeScript, use the [`heartbeat()`](https://typescript.temporal.io/api/namespaces/activity#heartbeat) function in your Activity implementation.
 
-To set Activity Heartbeat, use the `heartbeat()` function in your Activity implementation, and set `heartbeatTimeout` in your Workflow.
-
-```ts
-// activity implementation
-export async function example(sleepIntervalMs = 1000): Promise<void> {
+```typescript
+export async function myActivity(): Promise<void> {
   for (let progress = 1; progress <= 1000; ++progress) {
-    await sleep(sleepIntervalMs);
-    // record activity heartbeat
+    // Do something that will take some time
+    await sleep('1s');
+
+    // Send a heartbeat
     heartbeat();
   }
 }
-
-// ...
-
-// workflow code calling activity
-const { example } = proxyActivities<typeof activities>({
-  startToCloseTimeout: '1 hour',
-  heartbeatTimeout: '10s',
-});
 ```
 
-In the previous example, setting the Heartbeat informs the Temporal Server of the Activity's progress at regular intervals.
-If the Activity stalls or the Activity Worker becomes unavailable, the absence of Heartbeats prompts the Temporal Server to retry the Activity immediately, without waiting for `startToCloseTimeout` to complete.
-
-You can also add `heartbeatDetails` as a checkpoint to collect data about failures during the execution, and use it to resume the Activity from that point.
-
-The following example extends the previous sample to include a `heartbeatDetails` checkpoint.
-
-```ts
-export async function example(sleepIntervalMs = 1000): Promise<void> {
-  const startingPoint = activityInfo().heartbeatDetails || 1; // allow for resuming from heartbeat
-  for (let progress = startingPoint; progress <= 100; ++progress) {
-    await sleep(sleepIntervalMs);
-    heartbeat(progress);
-  }
-}
-```
+In addition to obtaining cancellation information, Heartbeats also support detail data that persists on the server for retrieval during Activity retry.
```

Comment:
Does persisting on the server mean that it becomes part of the Event History? If so, I think it would be good to clarify that.

And is it cancellation information or cancellation requests?

Reply:
> Does persisting on the server mean that it becomes part of the Event History? If so, I think it would be good to clarify that.

To be exact, it gets persisted to the Workflow's Mutable State, which is a very advanced concept. There's no corresponding encyclopedia entry to which I could link to… I think it would be preferable not to mention that.